src/metabase/sync/sync_metadata.clj

```diff
@@ -49,7 +54,14 @@
     (sync-util/run-sync-operation "sync" database sync-steps)))
 
 (s/defn sync-table-metadata!
-  "Sync the metadata for an individual `table` -- make sure Fields and FKs are up-to-date."
-  [table :- i/TableInstance]
-  (sync-fields/sync-fields-for-table! table)
-  (sync-fks/sync-fks-for-table! table))
+  "Sync the metadata for an individual `tbl` -- make sure Fields and FKs are up-to-date. Foreign keys are only
+  synced if the table's database's driver supports the `:foreign-keys` feature."
+  [tbl :- i/TableInstance]
+  (sync-fields/sync-fields-for-table! tbl)
+  (let [drv (-> tbl table/database driver.u/database->driver)]
```

Comment:
surprising that a bug like this has existing for so long

Reply:
ah, there's a check for this below. this is just an optimization it appears?

Reply:
Unsure what you mean by "below", here?  But I think that without checking whether the driver supports FKs, the `sync-fks-for-table!` will get called, and may fail (or simply do nothing, depending on how the JDBC driver behavior works).  I don't think it breaks sync or anything when that happens, since it will just capture that error status and continue doing the other tasks.  But it's a bit of an optimization, I suppose, so that it will only be attempted if it's thoeretically supported.

Reply:
https://github.com/metabase/metabase/blob/master/src/metabase/sync/fetch_metadata.clj#L24
This is what they eventually call and it's a no-op when the driver doesn't support it it seems. By below i meant further in the call stack.

Reply:
Ah, not sure how I missed that!  I'm also unsure of the circumstances in which I was seeing the exception from the JDBC driver pertaining to that feature not being supported.  Thanks; I have backed out the change to this file.